src/eko/output/struct.py

```diff
@@ -0,0 +1,502 @@
+            info.size = len(stream.getbuffer())
+            info.mtime = time.time()
+            info.mode = 436
+            #  info.uname = os.getlogin()
+            #  info.gname = os.getlogin()
+
+            with tarfile.open(self.path, "a") as tar:
+                tar.addfile(info, fileobj=stream)
+
+        self._operators[q2] = op
+
+    def __delitem__(self, q2: float):
+        """Drop operator from memory.
+
+        This method only drops the operator from memory, and it's not expected
+        to do anything else.
+
+        Autosave is done on set, and explicit saves are performed by the
+        computation functions.
+
+        If a further explicit save is required, repeat explicit assignment::
+
+            eko[q2] = eko[q2]
+
+        This is only useful if the operator has been mutated in place, that in
+        general should be avoided, since the operator should only be the result
+        of a full computation or a library manipulation.
+
+
+        Parameters
+        ----------
+        q2 : float
+            the value of :math:`Q^2` for which the corresponding operator
+            should be dropped
+
+        """
+        self._operators[q2] = None
+
+    @contextlib.contextmanager
+    def operator(self, q2: float):
+        try:
+            yield self[q2]
+        finally:
+            del self[q2]
+
+    @property
+    def Q2grid(self):
+        return np.array(list(self._operators))
+
+    def __iter__(self):
+        """Iterate over keys (i.e. Q2 values)
+
+        Yields
+        ------
+        float
+            q2 values
+
+        """
+        for q2 in self._operators:
+            yield q2
+
+    def items(self):
+        """Iterate operators, with minimal load.
+
+        Pay attention, this iterator:
+
+        - is not a read-only operation from the point of view of the in-memory
+          object (since the final result after iteration is no operator loaded)
+        - but it is a read-only operation from the point of view of the
+          permanent object on-disk
+
+        Yields
+        ------
+        tuple
+            couples of ``(q2, operator)``, loaded immediately before, unloaded
+            immediately after
+
+        """
+        for q2 in self.Q2grid:
+            yield q2, self[q2]
+            del self[q2]
+
+    def __contains__(self, q2: float) -> bool:
+        return q2 in self._operators
+
+    def approx(self, q2, rtol=1e-6, atol=1e-10) -> Optional[float]:
+        q2s = self.Q2grid
+        close = q2s[np.isclose(q2, q2s, rtol=rtol, atol=atol)]
+
+        if close.size == 1:
+            return close[0]
+        if close.size == 0:
+            return None
+        raise ValueError(f"Multiple values of Q2 have been found close to {q2}")
+
+    @staticmethod
+    def bootstrap(tdir: PathLike, theory: dict, operator: dict):
+        tdir = pathlib.Path(tdir)
+        (tdir / "theory.yaml").write_text(yaml.dump(theory), encoding="utf-8")
```

Comment:
same as before -> module constants

Reply:
do you mean `"theory.yaml"`, it might be a good idea. I agree about getting rid of duplicated strings (it was part of the idea of this module in the first place)

Reply:
Solved in 1b8bb42bfa5a9f98670ca007dd78eb67ee9c4dda